Documentation/hubble.rst

```diff
@@ -0,0 +1,100 @@
+****************
+Hubble internals
+****************
+
+.. note:: This documentation section is targeted at developers who are
+          interested in contributing to Hubble. For this purpose, it describes
+          Hubble internals.
+
+.. note:: This documentation covers the Hubble server (sometimes referred as
+          "Hubble embedded") and Hubble-relay components but does not cover the
+          Hubble UI and CLI.
+
+Hubble builds on top of Cilium and eBPF to enable deep visibility into the
+communication and behavior of services as well as the networking infrastructure
+in a completely transparent manner.
+
+One of the design goals of Hubble is to achieve all of this at large scale.
+Although the initial release of Hubble was a standalone component, as of Cilium
+1.8, Hubble server is embedded into the Cilium agent. This decision has been
+taken due to performance concerns and drift with the Hubble standalone approach
+as Hubble had to replicate a lot of the Cilium state in its process by calling
+Cilium API. This not only contributed to performance overhead but more
+importantly was the source of a lot of code duplication and maintenance burden.
+
+.. note:: This guide does not cover Hubble in standalone mode, which is
+          deprecated.
+
+Hubble Architecture
+===================
+
+Hubble exposes gRPC services from the Cilium process that allows clients to
+receive flows and other type of data.
+
+Hubble server
+-------------
+
+The Hubble server component implements two gRPC services. The **Peer service**,
+which is only served on a local Unix domain socket, and the **Observer service**
```

Comment:
nit: You introduce these in the order (1) peer then (2) observer but the sections are the reverse order. It's cleaner to have them follow the same order (ideally of importance, or knowledge you need to build up in order to proceed to the next section).